app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ShoppingListController < ApplicationController
+  def index
+    @recipe_foods = RecipeFood.all
+    @food = Food.includes(:recipe_foods).where(user_id: current_user.id)
+    @total_price = 0
+    # @recipe_foods.each do |recipe_food|
+    # @total_price += recipe_food.food.price * recipe_food.quantity
+    # end
+    displayed_names = []
+    @recipe_foods.each do |recipe_food|
+      unless displayed_names.include?(recipe_food.food.name)
+        displayed_names << recipe_food.food.name
+        @total_price += recipe_food.food.price * recipe_food.quantity
+      end
+    end
+    @displayed_names = displayed_names
+  end
```

Comment:
- [ ] Let's ensure the general shopping list renders (In the data entered chicken is missing):
  * renders the list of food that is missing for all recipes of the logged-in user.
  * counts the total food items and total price of the missing food.
  
  **Your General shopping list view:**
  
![image](https://github.com/Chandan-devs-tech/recipe-app/assets/12745474/356ad43e-65e0-43a1-91e2-307f3494ad6c)